### DISCUSSION TITLE
total rows / pages count

### DISCUSSION THREAD

**Discussion by marco-santulli** [Budibase/budibase#3909] [Help]
Hi,

is there a way to show the total number of rows in a table and possibly also the total number of pages next the the current page number?

M

**Comment by mjashanks** [comment]
There is no in-built way to do this, but you can build it up manually using a view.

So, 
1. Go to your table and create a view, named "Row Count"
2. Choose "Calculate", then "The Count of _id or AutoId"
3. Go into your design and add a new repeater block, set the data source to your `Row Count` view

Now you should be able to 
1. add the total rows binding, mine looks like : `{{ Row Count Repeater.My Table.value }}`
2. Calculate the number of pages, using JS: `return Math.ceil($("Row Count Repeater.My Table.value") / 8);` ("8" is your number of rows per page)

**Reply by marco-santulli** [reply]
Hi Mike,

thanks for the answer,

for those who are using an external data source, thee same can be achieved with a query that returns the number of rows.  Of course this becomes tricky with filters and searches. And also can't add the number of pages in the footer next to the current page, so it would be great if it will be added in the future in any case, low priority :)

M

**Reply by philsonchoi** [reply]
Would I have to use a custom query out of Budibase DB if I wanted to get the number of rows created by Current User? Or could views support that? Sorry for my newbie questions.
